### PR TITLE
not stop after startup failure

### PR DESCRIPTION
**Pull request by gaosaroma**
如果配置Ark变量 sofa.ark.auto.uninstall.when.failed.enable=false，那么安装失败不自动卸载；否则，将自动卸载

配置方式：
在基座中的 conf/ark/bootstrap.properties 中，配置 sofa.ark.auto.uninstall.when.failed.enable=false

## Summary by CodeRabbit

- **New Features**
  - Introduced auto-uninstallation functionality during biz installation and error handling.
  
- **Tests**
  - Added test methods and behavior mocking for various operations, improving test coverage and reliability.

- **Bug Fixes**
  - Enhanced error handling during biz installation with new constants and conditional checks.

### CODE REVIEW

File: sofa-ark-parent/core/api/src/main/java/com/alipay/sofa/ark/api/ArkClient.java [hunk 1]
```diff
@@ -206,12 +208,18 @@ private static ClientResponse doInstallBiz(File bizFile, String[] args, Map<Stri
...
             getLogger().error(response.getMessage(), throwable);
-            try {
-                biz.stop();
-            } catch (Throwable e) {
-                getLogger().error(String.format("UnInstall Biz: %s fail.", biz.getIdentity()), e);
-            } finally {
-                bizManagerService.unRegisterBizStrictly(biz.getBizName(), biz.getBizVersion());
+
+            boolean autoUninstall = Boolean.parseBoolean(ArkConfigs.getStringValue(
+                AUTO_UNINSTALL_ENABLE, "true"));
```
Comment (lvjing2) [inline sofa-ark-parent/core/api/src/main/java/com/alipay/sofa/ark/api/ArkClient.java:213]: 可以参考 ark.properties 的配置方式

Comment (gaosaroma) [reply sofa-ark-parent/core/api/src/main/java/com/alipay/sofa/ark/api/ArkClient.java:212]: 已复用基座的 conf/ark/bootstrap.properties

Comment (gaosaroma) [reply sofa-ark-parent/core/api/src/main/java/com/alipay/sofa/ark/api/ArkClient.java:212]: > 可以参考 ark.properties 的配置方式

已复用基座的 conf/ark/bootstrap.properties

File: sofa-ark-parent/core/spi/src/main/java/com/alipay/sofa/ark/spi/constant/Constants.java [hunk 2]
```diff
@@ -198,6 +198,8 @@ public class Constants {
     public final static String       EMBED_STATIC_BIZ_ENABLE                       = "sofa.ark.embed.static.biz.enable";
     public final static String       ACTIVATE_NEW_MODULE                           = "activate.new.module";
 
+    public final static String       AUTO_UNINSTALL_ENABLE                         = "sofa.ark.auto.uninstall.enable";
```
Comment (lvjing2) [inline sofa-ark-parent/core/spi/src/main/java/com/alipay/sofa/ark/spi/constant/Constants.java:201]: sofa.ark.auto.uninstall.when.failed.enable，这个是不是更能明白是失败时自动卸载？

**Review by lvjing2 — APPROVED** [review]
LGTM

**Pull request merged**